CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ set(CMAKE_MODULE_PATH
     ${CMAKE_MODULE_PATH}
     "${CMAKE_SOURCE_DIR}/cmake/Modules/")
 
-find_package(Boost REQUIRED COMPONENTS program_options system thread)
+find_package(Boost REQUIRED COMPONENTS program_options system thread regex)
```

Comment:
> Why to we need regex again?

No idea - you tell me :)

The linker was asking for that regex library (and CMake was missing in capturing it).

If no project is using RegEx, then there is some deeper issue here - for some reason linker is trying to link to the boost-regex and complaining when not found by default.